src/views/ModelPlan/TaskList/index.tsx

```diff
@@ -213,6 +231,77 @@ const TaskList = () => {
     );
   };
 
+  const [updateModelStatus] = useUpdateModelPlanMutation();
+
+  useEffect(() => {
+    if (modelStatus && !loading) {
+      const translatedStatus = statusConfig.options[modelStatus];
+
+      if (modelStatus.toUpperCase() !== status) {
+        updateModelStatus({
+          variables: {
+            id: modelID,
+            changes: {
+              status: modelStatus
+            }
+          }
+        })
+          .then(response => {
+            if (!response?.errors) {
+              setStatusMessage({
+                message: t('statusUpdateSuccess', {
+                  status: translatedStatus
+                }),
+                status: 'success'
+              });
+
+              params.delete('model-status');
+              history.replace({
+                search: params.toString()
+              });
+
+              refetch();
+            }
+          })
+          .catch(errors => {
+            setStatusMessage({
+              message: t('statusUpdateError', {
+                status: translatedStatus
+              }),
+              status: 'error'
+            });
+            params.delete('model-status');
+            history.replace({
+              search: params.toString()
+            });
+          });
```

Comment:
getting inaccurate error when i enter `?model-status=CANCEL`

![image](https://github.com/user-attachments/assets/49a88cb7-8c2f-4621-9ffd-45358def8022)

Reply:
CANCEL isn't a status

Reply:
We don't have translations for CANCEL and nor would we really ever have that passed

Reply:
oh right. i was testing a status that does not exist.
Nevertheless, the error banner says `there was an error updating the status to .` <--

Reply:
Right that will never happen, it also hasn't been mocked.  That text always runs through i18n translation.  I can ask Natasha but I dont thin kthis scenario would ever happen as it's coming from email link.  I'll see if I can get a general error from her.

Reply:
@garyjzhao Just ended up adding the raw value if attempting to add something other than a set status